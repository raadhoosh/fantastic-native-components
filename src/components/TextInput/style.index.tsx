import styled from "styled-components/native";
const Text = styled.Text`
color: ${(props: any) => props.theme.primary.main};
`;

export default Text;