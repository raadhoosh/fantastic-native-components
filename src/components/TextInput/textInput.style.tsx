import styled from "styled-components";
import { TextInput as RnTextInput } from "react-native";

function getColor(props: any) {
    let bgC = "#6c757d";
    let forC = "#fff";
    if (props.backgroundColor) {
        bgC = props.backgroundColor;
    } else {
        const color =
            (props.primary && "primary")
            || (props.secondary && "secondary")
            || (props.success && "success")
            || (props.info && "info")
            || (props.warning && "warning")
            || (props.danger && "danger");

        if (typeof (color) === "string") {
            bgC = props.theme[color].main;
            forC = props.theme[color].contrastText;
        }
    }
    return bgC    
}

const TextInput = styled(RnTextInput)`    
    background-color: ${(props: any) => props.backgroundColor ? props.backgroundColor : '#fff'};    
    border: 1px solid ${(props) => getColor(props)};          
    width: ${(props: any) => props.width ? props.width : "auto"};    
    border-radius: 3px;
    margin-bottom: 5px;
    padding: 5px 10px;
`;

export default TextInput;