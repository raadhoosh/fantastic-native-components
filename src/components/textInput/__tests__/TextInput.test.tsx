import TextInput from "../index";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

const createTestProps = (props: Object) => ({
  ...props,
});

describe("TextInput", () => {
  describe("rendering", () => {
    let component: ShallowWrapper;
    let props: Object;
    beforeEach(() => {
      props = createTestProps({
        color: '#f00',
        primary: true,
        secondary: true,
        success: true,        
      });
      component = shallow(<TextInput {...props}
        onChangeText={() => {
          alert('You tapped the Text!');
        }}
      />);
    });
    it("should render a <TextInput />", () => {
      const component = shallow(<TextInput {...props}
        onChangeText={() => {
          alert('You tapped the Text!');
        }}
      />);
      expect(component).toMatchSnapshot();
    });
  });
});
