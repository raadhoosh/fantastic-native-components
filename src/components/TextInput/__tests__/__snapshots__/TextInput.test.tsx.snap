// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput rendering should render a <TextInput /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    color="#f00"
    onChangeText={[Function]}
    primary={true}
    secondary={true}
    success={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "color": "#f00",
      "editable": true,
      "onChangeText": [Function],
      "primary": true,
      "secondary": true,
      "success": true,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "State": Object {
        "blurTextInput": [Function],
        "currentlyFocusedField": [Function],
        "focusTextInput": [Function],
      },
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "    
    background-color: ",
          [Function],
          ";    
    border: 1px solid ",
          [Function],
          ";          
    width: ",
          [Function],
          ";    
    border-radius: 3px;
    margin-bottom: 5px;
    padding: 5px 10px;
",
        ],
      },
      "displayName": "Styled(TextInput)",
      "getDefaultProps": [Function],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "color": "#f00",
        "editable": true,
        "onChangeText": [Function],
        "primary": true,
        "secondary": true,
        "success": true,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "State": Object {
          "blurTextInput": [Function],
          "currentlyFocusedField": [Function],
          "focusTextInput": [Function],
        },
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "    
    background-color: ",
            [Function],
            ";    
    border: 1px solid ",
            [Function],
            ";          
    width: ",
            [Function],
            ";    
    border-radius: 3px;
    margin-bottom: 5px;
    padding: 5px 10px;
",
          ],
        },
        "displayName": "Styled(TextInput)",
        "getDefaultProps": [Function],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
